src/test/java/net/finmath/marketdata/model/curves/NelsonSiegelSvenssonBondCalibrationTest.java

```diff
@@ -0,0 +1,145 @@
+package net.finmath.marketdata.model.curves;
+
+import net.finmath.marketdata.calibration.ParameterObject;
+import net.finmath.marketdata.calibration.Solver;
+import net.finmath.marketdata.model.AnalyticModel;
+import net.finmath.marketdata.model.AnalyticModelFromCurvesAndVols;
+import net.finmath.marketdata.model.bond.Bond;
+import net.finmath.marketdata.products.AnalyticProduct;
+import net.finmath.optimizer.SolverException;
+import net.finmath.time.Schedule;
+import net.finmath.time.ScheduleGenerator;
+import net.finmath.time.businessdaycalendar.BusinessdayCalendarExcludingTARGETHolidays;
+import org.junit.Test;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import java.util.stream.Collectors;
+
+//TODO
+//it is not clear why interpolated yield is so different from the actual yield. For exmaple for the first bond in the list (the one with maturity 0.1 years)
+//actual yield: 11.616161616161616, interpolated yield : -15.188921707979922, exampleInterpolatedYield: 7.187521869740281
+
+public class NelsonSiegelSvenssonBondCalibrationTest {
+
+	public static final String DISCOUNT_CURVE_NAME = "discountCurve-EUR";
+	public static final String ACT_365 = "ACT/365";
+	public static final String ANNUAL = "annual";
+	public static final LocalDate REFERENCE_DAY = LocalDate.of(2020, 11, 1);
+
+	/*
+	 * Test the calibration of a Nelson-Siegel-Svensson curve to a given set of bond.
+	 * Input values for the model are:
+	 *  - set of bonds' maturity,
+	 *  - coupon rates,
+	 *  - current prices
+	 * Then we calibrate the curve using current prices. The aim is to calibrate model in a way that interpolated yields are close enough to actual yields
+	 * @throws SolverException Thrown if the solver cannot find a solution.
+	 */
+	@Test
+	public void testBondNSSCurveCalibration() throws SolverException {
+		//maturities in years.
+		double[] maturity = new double[]{0.1, 0.3, 1.0, 2.0, 3.0, 4.0, 5.0, 7.0, 10.0, 20.0, 30.0};
+		//coupon rates of bonds, 1.5% for all of them
+		double[] rates = new double[]{0.015, 0.015, 0.015, 0.015, 0.015, 0.015, 0.015, 0.015, 0.015, 0.015, 0.015};
+		//current prices of bonds, current price for face value of 100. (i.e. 99 means that the bond is sold with a discount)
+		double[] prices = new double[]{99.0, 99.8, 101.0, 101.5, 102.0, 102.8, 103.0, 103.8, 103.9, 104.0, 104.2};

```

Comment:
The bond assumes a notion of 1.0. So prices should be 0.99, 0.998, 1.01,...

Reply:
I have fixed this below (in a stream.map(x -> x/100)), but maybe it should be fixed here